processor/src/advice/mod.rs

```diff
@@ -19,47 +68,53 @@ use vm_core::{
 ///    trees and can be used to provide Merkle paths.
 ///
 /// An advice provider can be instantiated from [ProgramInputs].
-pub struct AdviceProvider {
+#[derive(Debug, Clone, Default)]
+pub struct BaseAdviceProvider {
```

Comment:
I think `BaseAdviceProvider` is not super descriptive. Maybe it should something like `MemAdviceProvider` to indicate that it stores data in memory?

Also, I would probably move it out of the `mod.rs` into a separate module - e.g., `src/advice/mem_provider.rs`.

Reply:
That makes sense; done!